core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveIntegralCodec.java

```diff
@@ -308,15 +308,49 @@ public void decodeAndFillVector(byte[] pageData, ColumnVectorInfo vectorInfo, Bi
     private void fillVector(byte[] pageData, CarbonColumnVector vector, DataType vectorDataType,
         DataType pageDataType, int pageSize, ColumnVectorInfo vectorInfo, BitSet nullBits) {
       int rowId = 0;
+      CarbonColumnVector vectorColumn = null, childrenVectorInfo = null;
+      vectorColumn = vectorInfo.vector.getColumnVector();
+      if (vectorColumn != null) childrenVectorInfo = vectorColumn.getChildrenVector();
```

Comment:
Same code is added in AdaptiveFloatingCodec. Extract common code to a method and reuse